python/python_fastapi_server/routers/middleware.py

```diff
@@ -0,0 +1,25 @@
+"""
+Sets the request scheme to a value derived from the EXTERNALADDRESS env setting
+
+For use behind proxies
+"""
+import os
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+
+class FixSchemeMiddleware:
+    """Fix https scheme behind proxy"""
+
+    def __init__(self, app: ASGIApp):
+        self.app = app
+        self.scheme = None
+        external_address = os.environ["EXTERNALADDRESS"]
+        if external_address:
+            self.scheme = external_address[: external_address.index(":")]
```

Comment:
Maybe safer to use this?
https://docs.python.org/3/library/urllib.parse.html

Reply:
Done